docs/upgrade.html

```diff
@@ -42,6 +42,11 @@ <h5><a id="upgrade_250_notable" href="#upgrade_250_notable">Notable changes in 2
         enabled by default. You can continue to use TLSv1 and TLSv1.1 by explicitly enabling these in the configuration options
         <code>ssl.protocol</code> and <code>ssl.enabled.protocols</code>.
     </li>
+    <li>ZooKeeper has been upgraded to version 3.5.7.  This resolves bugs that could have impacted ZooKeeper upgrades as mentioned
+        in the 2.4.0 upgrade notes, so those comments do not apply to this version of Kafka.  ZooKeeper version 3.5.7 supports TLS-encrypted connectivity
```

Comment:
"so the workaround mentioned in 2.4 do not apply to this version of Kafka"

Reply:
@rondagostino since ZOOKEEPER-3056 is still required for any 3.4 -> 3.5.6 and above upgrade,  Can we repeat upgrade description like:
"ZooKeeper has been upgraded to 3.5.7. ZooKeeper upgrade from 3.4.X to 3.5.6 and above can fail if there are no snapshot files in 3.4 data directory. This usually happens in test upgrades where ZooKeeper 3.5.7 is trying to load an existing 3.4 data dir in which no snapshot file has been created. For more details about the issue please refer to ZOOKEEPER-3056. A fix is given in ZOOKEEPER-3056, which is to set snapshot.trust.empty=true config in zookeeper.properties before the upgrade."

Reply:
Thanks, @omkreddy.  I'm assuming the part about data loss potentially occurring with `snapshot.trust.empty=true` does not need to be here since https://issues.apache.org/jira/browse/ZOOKEEPER-3644 is fixed in 3.5.7.  I'm also assuming we do not need to mention it in the release notes.